src/Validator/hdtAssetValidator.cpp

```
@@ -180,17 +181,19 @@ namespace hdt
 	{
 		XmlRedundancyInfo result;
 
-		std::string bytes = readAllFile2(xmlPath.c_str());
-		if (bytes.empty())
+		// Analyse the same fully-expanded document the runtime and other validators see. A malformed
+		// pattern is reported by the XSD validator, so just skip redundancy analysis here.
+		PhysicsXmlSource src = readAndExpandPhysicsXml(xmlPath);
+		if (src.xml.empty() || !src.ok)
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)